new-packages/ts-project/__tests__/transitions.test.ts

```diff
@@ -3034,3 +3034,392 @@ test('should extract transition description (multi-line)', async () => {
       ]
     `);
 });
+test('should extract after transition (number delay)', async () => {
+  const tmpPath = await testdir({
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        initial: "foo",
+        states: {
+          foo: {
+            after: {
+              100: "bar",
+            },
+          },
+          bar: {},
+        },
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+  expect(replaceUniqueIds(project.extractMachines('index.ts')))
+    .toMatchInlineSnapshot(`
+    [
+      [
+        {
+          "blocks": {},
+          "data": {
+            "context": {},
+          },
+          "edges": {
+            "edge-0": {
+              "data": {
+                "actions": [],
+                "description": undefined,
+                "eventTypeData": {
+                  "delay": "100",
```

Comment:
I wonder - does Studio try to auto-detect if this is a number or a string?

Reply:
It does but for formatting and displaying purposes. The delay that's stored in digraph definition is always a string.